src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/FormLoginLdapFactory.php

```diff
@@ -34,9 +34,14 @@ protected function createAuthProvider(ContainerBuilder $container, $id, $config,
             ->replaceArgument(2, $id)
             ->replaceArgument(3, new Reference($config['service']))
             ->replaceArgument(4, $config['dn_string'])
+            ->replaceArgument(5, $config['search_dn'])
+            ->replaceArgument(6, $config['search_password'])
         ;
 
         if (!empty($config['query_string'])) {
+            if (empty($config['search_dn']) && empty($config['search_password'])) {
```

Comment:
is this correct?
also, we should not use empty when it can be avoided.
would this work (note the `||`)?
`if ('' === $config['search_dn'] || '' === $config['search_password']) {`

Reply:
This is correct we need both for this to work, the `||` does not work here, I agree that we should not use empty, I was only using the same kind of test that the query_string.

Reply:
> we need both for this

so we need `||` to me: if any is empty, this should trigger.

Reply:
you are right, didn't saw it like that.